python/lib/dependabot/python/version.rb

```diff
@@ -14,7 +14,7 @@ class Version < Gem::Version
       attr_reader :local_version
       attr_reader :post_release_version
 
-      VERSION_PATTERN = 'v?([0-9]+!)?[0-9]+[0-9a-zA-Z]*(?>\.[0-9a-zA-Z]+)*' \
+      VERSION_PATTERN = 'v?([1-9][0-9]*!)?[0-9]+[0-9a-zA-Z]*(?>\.[0-9a-zA-Z]+)*' \
```

Comment:
I guess this makes sense because there wouldn't be a version like `0!0.0.1`.
Probably doesn't matter but we could also make the new group a non-capturing group like the other one by adding `?>`.

Reply:
Oh I forgot to mention this! I grabbed the epoch portion of the [official regex](https://peps.python.org/pep-0440/#appendix-b-parsing-version-strings-with-regular-expressions). This might be a little too pedantic, having a leading 0 wouldn't matter, maybe I should switch it back to be more forgiving 🤔

Reply:
Ah! I didn't know that was available. Personally, I think what you have is fine.

Reply:
Yeah we should definitely adopt the official regexes. 

I started pulling in the Python regex with capture groups so we can get away from all the splitting and just use the groups, but saw it was going to change current behavior. I thought it would be best to fix epochs on the first pass.